GameOfRunes/GameOfRunes/Scenes/GameScene.swift

```diff
@@ -55,210 +49,29 @@ class GameScene: SKScene, GameSceneFacade {
             fatalError("Unable to load stage or/and avatar from GameStateMachine")
         }
 
-        gameEngine = GameEngine(gameScene: self, stage: stage, avatar: avatar)
+        let gameEngine = GameEngine(stage: stage,
```

Comment:
should instantiate within the root node

Reply:
Yup I can refactor into this, but the thing is that the PhysicsWorld's contactDelegate still has to be set from GameScene. Since the gameEngine inside RootRenderNode is of type GameEngineFacade, maybe just leave this outside of RootRenderNode so we can avoid the ugly typecasting? This is just a slight concern hahahaha